dds/idl/dds_generator.h

```diff
@@ -441,6 +441,73 @@ std::string wrapPrefix(AST_Type* type, WrapDirection wd)
   }
 }
 
+inline std::string string_type(AstTypeClassification::Classification cls)
+{
+  return be_global->language_mapping() == BE_GlobalData::LANGMAP_CXX11 ?
+    ((cls & AstTypeClassification::CL_WIDE) ? "std::wstring" : "std::string") :
+    (cls & AstTypeClassification::CL_WIDE) ? "TAO::WString_Manager" : "TAO::String_Manager";
+}
+
+inline std::string to_cxx_type(AST_Type* type, std::size_t& size)
+{
+  const AstTypeClassification::Classification cls = AstTypeClassification::classify(type);
+  if (cls & AstTypeClassification::CL_ENUM) {
+    size = 4;
+    return "ACE_CDR::ULong";
+  }
+  if (cls & AstTypeClassification::CL_STRING) {
+    return string_type(cls);
+  }
+  if (cls & AstTypeClassification::CL_PRIMITIVE) {
+    AST_Type* t = AstTypeClassification::resolveActualType(type);
+    AST_PredefinedType* p = dynamic_cast<AST_PredefinedType*>(t);
+    switch (p->pt()) {
+    case AST_PredefinedType::PT_long:
+      size = 4;
+      return "ACE_CDR::Long";
+    case AST_PredefinedType::PT_ulong:
+      size = 4;
+      return "ACE_CDR::ULong";
+    case AST_PredefinedType::PT_longlong:
+      size = 8;
+      return "ACE_CDR::LongLong";
+    case AST_PredefinedType::PT_ulonglong:
+      size = 8;
+      return "ACE_CDR::ULongLong";
+    case AST_PredefinedType::PT_short:
+      size = 2;
```

Comment:
What about using https://github.com/DOCGroup/ACE_TAO/blob/754e5a7a4d7c445d52b5376323bf19a6a206dc23/ACE/ace/CDR_Base.h#L77

Reply:
Except for the types, I don't think we're using anything from ACE's CDR in our serialization system, so it's not really a time to start. I'd rather us use the non-enum constants in `dds/DCPS/Serializer.h` anyway.